### ISSUE TITLE
Merchant can reappear on top of character

### ISSUE THREAD

**Issue by chris-torrence** [6502-workshop/NA#17]
In the castle, I was waiting for the food merchant to wake up, and I was apparently standing in the spot where he reappears. I waited for 6 hours, and then poof, he was on top of me. Moving made my character reappear.
<img width="1094" alt="screen shot 2018-11-04 at 10 24 43 am" src="https://user-images.githubusercontent.com/5461379/47975876-0c774400-e06d-11e8-9286-b446dbddc095.png">

**Comment by 6502-workshop** [comment]
Chris,

Thanks. What do you think of a work around, turning it into a joke by printing text like "Hey! Get outta my way!"


My Notes
-----------

*This didn't happen in Ultima V because "waiting" was only permitted in beds, which was really annoying if you rolled into town and the shop was closed.

*To fix we'd need to detect that NPC X,Y = player X,Y during NPC.INIT. If true then, adjust NPC X,Y. Would have to run adjusted X/Y through collision controls. If blocked, then what? An alternate X,Y would take a bunch of memory, disk, and potentially introduce bugs unless it could be a separate data structure (not sure).

*The building routines module which NPC.INIT resides in is out of memory. We'd need to offload a portion of the building routines into main game engine memory, if any is available.

**Comment by chris-torrence** [comment]
Yeah, I don't think it's a huge deal - it's just because I happened to be on that exact square. I like the "joke" idea - sort of like the NPCs that yell at you when you get too close to them.

**Comment by chris-torrence** [comment]
Yes, the NPCs have yelled at me. It was really funny!

**Comment by 6502-workshop** [comment]
Awesome! I really enjoyed working that feature into the NPCs pathfinder algorithm, glad to know it’s coming up during gameplay.

They used to say “beat it or I’ll call the brute squad!” :-)

But I thought that was pushing it.

**Comment by chris-torrence** [comment]
No, it's perfect the way it is. I also like the "quit horsing around!". Cute.

**Comment by 6502-workshop** [comment]
Thanks for letting me know on that too. The horse bit and the pushy NPCs are good examples of the kind of humor I’ve been envisioning through the game. Hopefully the dark tone will hold up too. I think it will as long as the humor is sprinkled in here and there.
